src/dune/expander.ml

```diff
@@ -322,11 +342,6 @@ let expand_and_record acc ~map_exe ~dep_kind ~expansion_kind
     | Static -> fun _ -> User_error.raise ~loc [ cannot_be_used_here pform ]
     | Dynamic -> Resolved_forms.add_ddep acc expansion
   in
-  let add_fail =
-    match expansion_kind with
-    | Static -> fun _ (f : fail) -> f.fail ()
```

Comment:
This clause was the cause of the error. It's too soon to raise the error at this stage, we should instead prefix the rule itself with the error.

Reply:
Makes sense